oauth/src/test/java/pl/edu/icm/unity/oauth/as/TokenSignerTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2013 ICM Uniwersytet Warszawski All rights reserved.
+ * See LICENCE.txt file for licensing information.
+ */
+
+package pl.edu.icm.unity.oauth.as;
+
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+import eu.unicore.util.configuration.ConfigurationException;
+import pl.edu.icm.unity.oauth.as.OAuthASProperties;
+
+public class TokenSignerTest
+{
+	
+	@Test(expected = ConfigurationException.class)
+	public void shouldThrowConfigExceptionWhenEmptyHMACsecret()
+	{
+		OAuthASProperties config = OAuthTestUtils.getConfig();
+		config.setProperty(OAuthASProperties.SIGNING_ALGORITHM,
+				OAuthASProperties.SigningAlgorithms.HS512.toString());
+		new TokenSigner(config, new MockPKIMan());
+	}
+	
+	@Test(expected = ConfigurationException.class)
+	public void shouldThrowConfigExceptionWhenToShortHMACsecret()
+	{
+		OAuthASProperties config = OAuthTestUtils.getConfig();
+		config.setProperty(OAuthASProperties.SIGNING_SECRET,
+				"b8e7ae12510bdfb1812e463a7f086122cf37e4f7");
+		config.setProperty(OAuthASProperties.SIGNING_ALGORITHM,
+				OAuthASProperties.SigningAlgorithms.HS512.toString());
+		new TokenSigner(config, new MockPKIMan());
+	}
+
+	@Test
+	public void shouldCreateHMACTokenSigner()
+	{
+		OAuthASProperties config = OAuthTestUtils.getConfig();
+		config.setProperty(OAuthASProperties.SIGNING_SECRET,
+				"b8e7ae12510bdfb1812e463a7f086122cf37e4f7");
+		config.setProperty(OAuthASProperties.SIGNING_ALGORITHM,
+				OAuthASProperties.SigningAlgorithms.HS256.toString());
+		try
```

Comment:
In such cases (this and the last case) is better expect something from the created TokenSigner object. And what is more important you can declare method to throws the exception - it will fail with error, what should be good enough to understand the problem.
This code as is here is not very helpful when failing as you won't see the exception itself.